server/issue.go

```diff
@@ -625,12 +625,16 @@ func httpAPIAttachCommentToIssue(ji Instance, w http.ResponseWriter, r *http.Req
 		conf := ji.GetPlugin().getConfig()
 		extraText := ""
 		for _, fileId := range post.FileIds {
-			mattermostName, jiraName, e := client.AddAttachment(api, attach.IssueKey, fileId, conf.maxAttachmentSize)
+			mattermostName, jiraName, mime, e := client.AddAttachment(api, attach.IssueKey, fileId, conf.maxAttachmentSize)
 			if e != nil {
 				notifyOnFailedAttachment(ji, mattermostUserId, attach.IssueKey, e, "file: %s", mattermostName)
 			}
 
-			extraText += "\n\nAttachment: !" + jiraName + "!"
+			if isImageMIME(mime) || isEmbbedableMIME(mime) {
+				extraText += "\n\nAttachment: !" + jiraName + "!"
+			} else {
+				extraText += "\n\nAttachment: [^" + jiraName + "]"
```

Comment:
Can you explain what's going on between these two syntax's @larkox?

Reply:
Sure. `!jiraName!` embeds the `jiraName` attachment in the comment.
`[^jiraName]` creates a link to the `jiraName` attachment.
Non-embeddable files like the txt from the issue should be just linked, while other files like images work better as embedded in the comment.

Reply:
I originally thought this was Mattermost markdown but reading again, it seems this is Jira markdown. Makes more sense now :+1: